### DISCUSSION TITLE
Start debug server with python option -bb?

### DISCUSSION THREAD

**Discussion by dstromberg** [pallets/flask#4826] [Q&A]
I'm working on porting a somewhat large python project from cpython 2.7 to cpython 3.10.  I know, it's a little late.

The strings-vs-bytes thing is, unsurprisingly, the largest issue.

I'd like to run our debug flask with cpython's -bb option to turn str<->bytes comparisons and implicit conversions into errors.

But if I run:
`/usr/local/bin/tact-python -bb /app/app/run.py runserver -h 0.0.0.0 -p 5000`
(tact-python is just a symlink to cpython 3.10)
If I do that, then I get:

```
$ docker exec -it title_pub_api /bin/bash
cmd output started 2022 Thu Sep 22 02:14:03 PM PDT
tact@tact_pub_api:/app$ ps axf
  PID TTY      STAT   TIME COMMAND
 1102 pts/0    Ss     0:00 /bin/bash
 1115 pts/0    R+     0:00  \_ ps axf
    1 ?        Ssl    0:03 /usr/local/bin/tact-python -bb /app/app/run.py runserver -h 0.0.0.0 -p 5000
   18 ?        Sl     0:19 /usr/local/bin/tact-python /app/app/run.py runserver -h 0.0.0.0 -p 5000

```
...and I'm not at all confident that the API (served by Flask) is checking for bytes<->str issues.  At least, I'm not getting any tracebacks that look like it is, and that pid 18 doesn't have the -bb.

Is there a way to get Flask's debug server to pass -bb to python (tact-python)?

I don't see a way of doing the same thing as -bb using an environment variable, BTW.

Thanks!

**Comment by dstromberg** [comment]
It just occurred to me I might be able to ditch the explicit /usr/local/bin/tact-python and trust #!/usr/local/bin/tact-python -bb

That might cause problems for debugpy later, but for now I don't really care about that.

**Comment by davidism** [comment]
You can run `flask` as a module.

```
$ python -bb -m flask
```

This is the case for most packages that provide entry points, not only Flask.
